### ISSUE TITLE
JUnit reporter does not filter control characters

### ISSUE THREAD

**Issue by correctmost** [pylint-dev/pylint#11452]
### Bug description

The JUnit reporter can produce invalid XML when checking files with control characters.

The ElementTree code does not prevent this from happening: https://github.com/python/cpython/issues/49416

### Configuration

```ini

```

### Command used

```shell
pylint --output-format=junit tests/functional/b/bad_char/bad_char_esc.py | xmllint -
```

### Pylint output

```
# xmllint output

-:6: parser error : PCDATA invalid Char value 27
tests/functional/b/bad_char/bad_char_esc.py:5:4:--&gt;t;---</failure>
                                                      ^
-:7: parser error : PCDATA invalid Char value 27
      <system-out>tests/functional/b/bad_char/bad_char_esc.py:5:4:--&gt;t;---
                                                                        ^
-:9: parser error : PCDATA invalid Char value 27
tests/functional/b/bad_char/bad_char_esc.py:5:4:--&gt;t;---</system-err>
```

### Expected behavior

Valid XML is produced

### Pylint version

```shell
ea90985a
Python 3.14.7
```

### Additional dependencies

```python

```

**Comment by Pierre-Sassoulas** [comment]
The only way for you to be more efficient that that would be to fuzz directly on opened PR 🧐
